x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -336,6 +337,14 @@ private void finalise(int slot, AutoFollowResult result) {
                                                     List<String> followedIndexUUIDs) {
             List<Index> leaderIndicesToFollow = new ArrayList<>();
             for (IndexMetaData leaderIndexMetaData : leaderClusterState.getMetaData()) {
+                // We should not automatically follow a leader index that is also a follow index and
+                // that both leader and follow index are in the same cluster otherwise this can
+                // result into an index creation explosion.
+                if (leaderIndexMetaData.getCustomData(Ccr.CCR_CUSTOM_METADATA_KEY) != null &&
+                    leaderClusterState.getClusterName().equals(followerClusterState.getClusterName())) {
```

Comment:
I am not sure if we should use the cluster name since the local and remote cluster can have the same name (I might be wrong here). Should we check if the cluster alias is "_local_"?

Reply:
> I am not sure if we should use the cluster name since the local and remote cluster can have the same name (I might be wrong here).

True, that is possible, although I think must users would have different names for different clusters.

> Should we check if the cluster alias is "local"?

Yes, I think that does the trick too and that is the actual scenario we do want the auto follow coordinator to not auto follow follower indices. I pushed this: df2093930d0b25f5308c7be5e1b1acb27a64b0ec